Pocket/pocket-ios
# feat: add url via manual input

## Summary
Add UI to allow user to add a URL to their Pocket

## References 
https://getpocket.atlassian.net/browse/IN-787

## Implementation Details
Added a new SwiftUI view and Model
The Model verifies the URL by simply attempting to use the user input as a URL object.

4 new analytics events have been created;

1. saves.addItem.open // Called when the AddSavedItem view is created
2. saves.addItem.dismiss // Called when the user taps either of the close buttons (but not when the view self-dismisses after a successful save)
3. saves.addItem.success // When the user successfully saves an item to their Pocket
4. saves.addItem.fail // When the Model rejects user input as invalid

Unit and UI Tests have been added.
The UI tests ensure the flows work as expected.
The Unit test bombards the URL validation code with good and bad URL examples.

## Test Steps
Open the app
Navigate to Saves
Tap the + button in the top right
Enter some text into the URL field
A valid URL should save and the view self-dismisses
An invalid URL should trigger an inline error message.

## PR Checklist:

## Screenshots
![Simulator Screen Recording - iPhone 14 - 2023-06-15 at 11 01 02](https://github.com/Pocket/pocket-ios/assets/8590135/ee3eb048-303a-4d4c-aaef-fde43427b44a)
